codegenerator/cli/templates/dynamic/codegen/src/Types.res.hbs

```diff
@@ -336,24 +298,28 @@ module HandlerTypes = {
 
     let noopLoader = _ => Promise.resolve()
 
-    let getLoaderHandler = (t: t<'eventArgs>) =>
-      t.loaderHandler->(
-        Utils.magic: option<loaderHandler<'eventArgs, Internal.loaderReturn, eventFilter>> => option<
-          loaderHandler<Internal.eventParams, Internal.loaderReturn, 'eventFilter>,
-        >
-      )
-
-    let getLoader = (t: t<'eventArgs>) => t->getLoaderHandler->Belt.Option.flatMap(lh => 
-      if lh.loader === noopLoader->(Utils.magic: loader<'eventArgs, ()> => loader<Internal.eventParams, Internal.loaderReturn>) {
-        None
-      } else {
-        Some(lh.loader)
+    let getLoader = (t: t<'eventArgs>) => 
+      switch t.loaderHandler {
+        | Some({loader}) => {
+          let loader = loader->(Utils.magic: loader<'eventArgs, Internal.loaderReturn> => Internal.loader)
```

Comment:
Does rescript optimise this variable allocation? Otherwise maybe it could be inlined? Could also just change the handlerLoader type to actually have an optional loader field rather than all this type casting.

Reply:
Yes, when you do casting it doesn't create an additional variable allocation. For the compiler it's still the same variable.

Reply:
> Could also just change the handlerLoader type to actually have an optional loader field rather than all this type casting.

Sounds like the best option.

Reply:
> > Could also just change the handlerLoader type to actually have an optional loader field rather than all this type casting.
> 
> Sounds like the best option.

We definitely want the user to get "undefined" as the loaderReturn type in typescript as well so we must just make sure that works.